### PR TITLE
Routine maintenance 2023-07

### LINKED ISSUE #32: Dependency Dashboard
This issue lists Renovate updates and detected dependencies. Read the [Dependency Dashboard](https://docs.renovatebot.com/key-concepts/dashboard/) docs to learn more.<br>[View this repository on the Mend.io Web Portal](https://developer.mend.io/github/LaunchPadLab/lp-redux-utils).

## Repository Problems

These problems occurred while renovating this repository. [View logs](https://developer.mend.io//github/LaunchPadLab/lp-redux-utils).

 - ⚠️ WARN: Using npm packages for Renovate presets is now deprecated. Please migrate to repository-based presets instead.

## Open

The following updates have all been created. To force a retry/rebase of any, click on a checkbox below.

 - [ ] [Update actions/checkout action to v7](../pull/56)
 - [ ] [Update actions/setup-node action to v6](../pull/53)
 - [ ] [Update babel monorepo to v8](../pull/57)
 - [ ] [Update dependency husky to v9](../pull/44)
 - [ ] [Update dependency redux to v5](../pull/43)
 - [ ] [Update eslint monorepo to v10](../pull/46)
 - [ ] [Update jest monorepo to v30](../pull/50)
 - [ ] [Update Node.js to v24](../pull/55)
 - [ ] **Click on this checkbox to rebase all open PRs at once**

## Detected Dependencies

<details><summary>github-actions (1)</summary>
<blockquote>

<details><summary>.github/workflows/publish.yml (3)</summary>

 - `actions/checkout v3` → [Updates: `v7`]
 - `actions/setup-node v3` → [Updates: `v6`]
 - `node 18.x` → [Updates: `24.x`]

</details>

</blockquote>
</details>

<details><summary>npm (1)</summary>
<blockquote>

<details><summary>package.json (11)</summary>

 - `lodash ^4.17.4` → [Updates: `^4.17.4`]
 - `@babel/cli ^7.22.9` → [Updates: `^8.0.0`]
 - `@babel/core ^7.22.9` → [Updates: `^7.22.9`]
 - `@launchpadlab/babel-preset ^2.1.1`
 - `@launchpadlab/eslint-config ^3.0.1`
 - `documentation ^14.0.0`
 - `eslint ^8.0.0` → [Updates: `^10.0.0`]
 - `husky ^8.0.0` → [Updates: `^9.0.0`]
 - `jest ^29.0.0` → [Updates: `^30.0.0`]
 - `redux ^4.0.0` → [Updates: `^5.0.0`]
 - `redux >=2.0.0`

</details>

</blockquote>
</details>

<details><summary>nvm (1)</summary>
<blockquote>

<details><summary>.nvmrc (1)</summary>

 - `node 18` → [Updates: `24`]

</details>

</blockquote>
</details>

<details><summary>renovate-config (1)</summary>
<blockquote>

<details><summary>.renovaterc</summary>

</details>

</blockquote>
</details>

---

- [ ] Check this box to trigger a request for Renovate to run again on this repository

### PR DESCRIPTION
**Pull request by chawes13**
Resolves #32 

Adds auto-publish GH action (trial run for other repos)

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      - name: Check out latest code
+        uses: actions/checkout@v3
+      - name: Set up node
+        uses: actions/setup-node@v3
+        with:
+          node-version: '18.x'
+          registry-url: 'https://registry.npmjs.org'
+          scope: '@launchpadlab'
+      - name: Install dependencies
+        run: HUSKY=0 yarn --frozen-lockfile
```
Comment (chawes13) [inline .github/workflows/publish.yml:19]: Setting `HUSKY` to 0 will skip the install of precommit hooks (https://typicode.github.io/husky/guide.html#with-env-variables)

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+name: Publish
```
Comment (chawes13) [inline .github/workflows/publish.yml:1]: I moved this from Travis to a GH action for a few reasons:
- Mitigates vendor lock-in (i.e., we're more likely to stay on GitHub than on Travis)
- Encourages good etiquette on creating Releases (i.e., won't publish to npm without a release created)
- Officially supported Slack notification integrations
- I wanted to test out GH actions

File: package.json [hunk 3]
```diff
@@ -10,8 +10,9 @@
     "clean": "rm -rf lib",
     "docs": "documentation build src/index.js -f md -o docs.md",
     "lint": "eslint src --max-warnings=0",
-    "prepublish": "yarn run lint && yarn run clean && yarn run build",
```
Comment (chawes13) [inline package.json:13]: `prepublish` was deprecated and split into `prepare` and `prepublishOnly`

File: package.json [hunk 4]
```diff
@@ -30,25 +31,20 @@
...
+    "@babel/cli": "^7.22.9",
+    "@babel/core": "^7.22.9",
     "@launchpadlab/babel-preset": "^2.1.1",
-    "@launchpadlab/eslint-config": "^2.7.0",
-    "documentation": "^12.1.4",
-    "eslint": "^6.8.0",
-    "husky": "^4.2.3",
-    "jest": "^25.1.0",
+    "@launchpadlab/eslint-config": "^3.0.1",
+    "documentation": "^14.0.0",
```
Comment (chawes13) [inline package.json:38]: The only change that affected this library was the change from `-` to `*` for bullet points

File: .github/workflows/publish.yml (hunk 2 above)
Comment (chawes13) [reply .github/workflows/publish.yml:1]: The fun (read: not so fun) bit about actions is you can't actually test them until they're merged. I added this to a private test repo on chawes13 and iterated a little bit so I'm _reasonably_ confident that this'll work as expected 🙂

Comment (bhennes2) [reply .github/workflows/publish.yml:1]: 👌

**Review by bhennes2 — APPROVED** [review]
LGTM!

**Pull request merged**